storage/src/adb/any/mod.rs

```diff
@@ -68,6 +70,31 @@ pub struct Config<T: Translator> {
     pub pool: Option<ThreadPool>,
 }
 
+/// Configuration for syncing an [Any] to a pruned target state.
+pub struct SyncConfig<E: RStorage + Metrics, K: Array, V: Array, T: Translator, D: Digest> {
+    /// Database configuration.
+    pub db_config: Config<T>,
+
+    /// The [Any]'s log of operations.
+    /// This log is expected to be pruned to the `pruned_to_loc` boundary
+    /// and contain all subsequent operations.
+    pub log: Journal<E, Operation<K, V>>,
+
+    /// The location in the [Any] up to which operations have been pruned.
+    /// This serves as both the log pruning boundary and the inactivity floor.
+    /// Everything before this location is considered pruned/inactive.
+    pub pruned_to_loc: u64,
+
+    /// The pinned nodes the MMR needs at the pruning boundary given by
+    /// `pruned_to_loc`, in the order specified by [Proof::nodes_to_pin].
+    pub pinned_nodes: Vec<D>,
+
+    /// The maximum number of operations to keep in memory
+    /// before committing the database while applying operations.
+    /// Higher value will cause more memory usage during sync.
+    pub max_ops_in_memory: usize,
```

Comment:
nit: easier to probably just call this "batch size"?

Reply:
Renamed `apply_batch_size` and renamed the other field to `fetch_batch_size` to be parallel